## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 461519`
Commit: `012283fada96e7a76f1389bcf0227b48409c0b4c`

---
### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python313Packages.wagtail-localize</li>
    <li>python313Packages.wagtail-localize.dist</li>
    <li>python313Packages.wagtail-modeladmin</li>
    <li>python313Packages.wagtail-modeladmin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.django-modelsearch</li>
    <li>python312Packages.django-modelsearch.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>python313Packages.django-modelsearch</li>
    <li>python313Packages.django-modelsearch.dist</li>
    <li>python313Packages.wagtail</li>
    <li>python313Packages.wagtail-factories</li>
    <li>python313Packages.wagtail-factories.dist</li>
    <li>python313Packages.wagtail.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python313Packages.wagtail-localize</li>
    <li>python313Packages.wagtail-localize.dist</li>
    <li>python313Packages.wagtail-modeladmin</li>
    <li>python313Packages.wagtail-modeladmin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.django-modelsearch</li>
    <li>python312Packages.django-modelsearch.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>python313Packages.django-modelsearch</li>
    <li>python313Packages.django-modelsearch.dist</li>
    <li>python313Packages.wagtail</li>
    <li>python313Packages.wagtail-factories</li>
    <li>python313Packages.wagtail-factories.dist</li>
    <li>python313Packages.wagtail.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python313Packages.wagtail-localize</li>
    <li>python313Packages.wagtail-localize.dist</li>
    <li>python313Packages.wagtail-modeladmin</li>
    <li>python313Packages.wagtail-modeladmin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.django-modelsearch</li>
    <li>python312Packages.django-modelsearch.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>python313Packages.django-modelsearch</li>
    <li>python313Packages.django-modelsearch.dist</li>
    <li>python313Packages.wagtail</li>
    <li>python313Packages.wagtail-factories</li>
    <li>python313Packages.wagtail-factories.dist</li>
    <li>python313Packages.wagtail.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.wagtail-localize</li>
    <li>python312Packages.wagtail-localize.dist</li>
    <li>python312Packages.wagtail-modeladmin</li>
    <li>python312Packages.wagtail-modeladmin.dist</li>
    <li>python313Packages.wagtail-localize</li>
    <li>python313Packages.wagtail-localize.dist</li>
    <li>python313Packages.wagtail-modeladmin</li>
    <li>python313Packages.wagtail-modeladmin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.django-modelsearch</li>
    <li>python312Packages.django-modelsearch.dist</li>
    <li>python312Packages.wagtail</li>
    <li>python312Packages.wagtail-factories</li>
    <li>python312Packages.wagtail-factories.dist</li>
    <li>python312Packages.wagtail.dist</li>
    <li>python313Packages.django-modelsearch</li>
    <li>python313Packages.django-modelsearch.dist</li>
    <li>python313Packages.wagtail</li>
    <li>python313Packages.wagtail-factories</li>
    <li>python313Packages.wagtail-factories.dist</li>
    <li>python313Packages.wagtail.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.wagtail-modeladmin</summary>
<pre>    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/zsiywj0zzab933hlishbxcjfha2x1ipb-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

======================================================================
FAIL: test_post_invalid (wagtail_modeladmin.test.tests.test_simple_modeladmin.TestEditView.test_post_invalid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File &quot;/build/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/zsiywj0zzab933hlishbxcjfha2x1ipb-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 2.841s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
<details>
<summary>python313Packages.wagtail-modeladmin</summary>
<pre>  File &quot;/build/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/9nyl4m8xgbcpcny6mry26fm5i67ik2b3-python3.13-django-4.2.26/lib/python3.13/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
    ~~~~~~~~~~~~~~~~^
        real_count,
        ^^^^^^^^^^^
    ...&lt;3 lines&gt;...
        % (real_count, text_repr, count),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 3.991s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.wagtail-modeladmin</summary>
<pre>    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/mzj3lpa0k09cy7s11jnr7fjjcfwjrivq-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

======================================================================
FAIL: test_post_invalid (wagtail_modeladmin.test.tests.test_simple_modeladmin.TestEditView.test_post_invalid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File &quot;/build/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/mzj3lpa0k09cy7s11jnr7fjjcfwjrivq-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 6.336s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
<details>
<summary>python313Packages.wagtail-modeladmin</summary>
<pre>  File &quot;/build/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/hksb9a079w92nfvqa7cc4yzf4dd09wls-python3.13-django-4.2.26/lib/python3.13/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
    ~~~~~~~~~~~~~~~~^
        real_count,
        ^^^^^^^^^^^
    ...&lt;3 lines&gt;...
        % (real_count, text_repr, count),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 6.024s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.wagtail-modeladmin</summary>
<pre>    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/48g8x2lqdlv8cp69954nqnzvj0mbzqd0-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

======================================================================
FAIL: test_post_invalid (wagtail_modeladmin.test.tests.test_simple_modeladmin.TestEditView.test_post_invalid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File &quot;/nix/var/nix/builds/nix-21224-3217970519/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/48g8x2lqdlv8cp69954nqnzvj0mbzqd0-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 3.068s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
<details>
<summary>python313Packages.wagtail-modeladmin</summary>
<pre>  File &quot;/nix/var/nix/builds/nix-25578-114025397/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/mihg6s6xki8aa7nb5x6r8awr7fby3m2c-python3.13-django-4.2.26/lib/python3.13/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
    ~~~~~~~~~~~~~~~~^
        real_count,
        ^^^^^^^^^^^
    ...&lt;3 lines&gt;...
        % (real_count, text_repr, count),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 3.205s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.wagtail-modeladmin</summary>
<pre>    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/mq3ycxg5cbv3lpyr0jrw7y1i3jz13jkz-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

======================================================================
FAIL: test_post_invalid (wagtail_modeladmin.test.tests.test_simple_modeladmin.TestEditView.test_post_invalid)
----------------------------------------------------------------------
Traceback (most recent call last):
  File &quot;/nix/var/nix/builds/nix-95766-518446848/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
  File &quot;/nix/store/mq3ycxg5cbv3lpyr0jrw7y1i3jz13jkz-python3.12-django-4.2.26/lib/python3.12/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 5.078s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
<details>
<summary>python313Packages.wagtail-modeladmin</summary>
<pre>  File &quot;/nix/var/nix/builds/nix-26330-1043523202/source/wagtail_modeladmin/test/tests/test_simple_modeladmin.py&quot;, line 748, in test_post_invalid
    self.assertContains(response, &quot;error-message&quot;, count=1)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/r9dc0jkv79k5ghyffw3zf4fz7jcb53hg-python3.13-django-4.2.26/lib/python3.13/site-packages/django/test/testcases.py&quot;, line 652, in assertContains
    self.assertEqual(
    ~~~~~~~~~~~~~~~~^
        real_count,
        ^^^^^^^^^^^
    ...&lt;3 lines&gt;...
        % (real_count, text_repr, count),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: 3 != 1 : Found 3 instances of &#x27;error-message&#x27; in response (expected 1)

----------------------------------------------------------------------
Ran 122 tests in 1.978s

FAILED (failures=2)
Destroying test database for alias &#x27;default&#x27;...</pre>
</details>
</details>

--- a/pkgs/development/python-modules/django-modelcluster/default.nix
+++ b/pkgs/development/python-modules/django-modelcluster/default.nix
@@ -1,22 +1,27 @@
 {
   lib,
   buildPythonPackage,
-  django-taggit,
-  django,
   fetchFromGitHub,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  django,
+  pytz,
+
+  # optional-dependencies
+  django-taggit,
+
+  # tests
   pytest-django,
   pytestCheckHook,
-  pythonOlder,
-  pytz,
-  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "django-modelcluster";
   version = "6.4";
   pyproject = true;
-
-  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "wagtail";
@@ -44,10 +49,10 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "modelcluster" ];
 
-  meta = with lib; {
+  meta = {
     description = "Django extension to allow working with 'clusters' of models as a single unit, independently of the database";
     homepage = "https://github.com/torchbox/django-modelcluster/";
-    changelog = "https://github.com/wagtail/django-modelcluster/blob/v${version}/CHANGELOG.txt";
-    license = licenses.bsd2;
+    changelog = "https://github.com/wagtail/django-modelcluster/blob/${src.tag}/CHANGELOG.txt";
+    license = lib.licenses.bsd2;
   };
 }

--- a/pkgs/development/python-modules/django-modelsearch/default.nix
+++ b/pkgs/development/python-modules/django-modelsearch/default.nix
@@ -1,0 +1,54 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  django,
+  django-tasks,
+
+  # tests
+  pytest-django,
+  pytestCheckHook,
+  django-modelcluster,
+}:
+
+buildPythonPackage rec {
+  pname = "modelsearch";
+  version = "1.1.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "wagtail";
+    repo = "django-modelsearch";
+    tag = "v${version}";
+    hash = "sha256-tjwVepI9mdrMbTtxfe6yNUrSHWKndGxv2lJ8AfyNcr0=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    django
+    django-modelcluster
+    django-tasks
+  ];
+
+  pythonImportsCheck = [ "modelsearch" ];
+
+  # django.core.exceptions.ImproperlyConfigured: Requested setting INSTALLED_APPS, but settings are not configured.
+  # You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
+  doCheck = false;
+
+  meta = {
+    description = "Index Django Models with Elasticsearch or OpenSearch and query them with the ORM";
+    homepage = "https://github.com/wagtail/django-modelsearch";
+    changelog = "https://github.com/wagtail/django-modelsearch/releases/tag/${src.tag}";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ GaetanLepage ];
+  };
+}

--- a/pkgs/development/python-modules/wagtail/default.nix
+++ b/pkgs/development/python-modules/wagtail/default.nix
@@ -1,7 +1,7 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitHub,
 
   # build-system
   setuptools,
@@ -12,6 +12,7 @@
   django,
   django-filter,
   django-modelcluster,
+  django-modelsearch,
   django-taggit,
   django-tasks,
   django-treebeard,
@@ -31,15 +32,14 @@
 
 buildPythonPackage rec {
   pname = "wagtail";
-  version = "7.1.1";
+  version = "7.2";
   pyproject = true;
 
-  # The GitHub source requires some assets to be compiled, which in turn
-  # requires fixing the upstream package lock. We need to use the PyPI release
-  # until https://github.com/wagtail/wagtail/pull/13136 gets merged.
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-e90eWww0VDeYXAHwp/YKYX5114jzfH2DlVj05qElGvk=";
+  src = fetchFromGitHub {
+    owner = "wagtail";
+    repo = "wagtail";
+    tag = "v${version}";
+    hash = "sha256-o/4jn32ffR3BPVNwtFKJ6PowXYi7SpjBqghdeZIl5tM=";
   };
 
   build-system = [
@@ -54,6 +54,7 @@ buildPythonPackage rec {
     django
     django-filter
     django-modelcluster
+    django-modelsearch
     django-taggit
     django-tasks
     django-treebeard

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4083,6 +4083,8 @@ self: super: with self; {
 
   django-modelcluster = callPackage ../development/python-modules/django-modelcluster { };
 
+  django-modelsearch = callPackage ../development/python-modules/django-modelsearch { };
+
   django-modeltranslation = callPackage ../development/python-modules/django-modeltranslation { };
 
   django-mptt = callPackage ../development/python-modules/django-mptt { };

